include/seqan3/range/container/small_string.hpp

```diff
@@ -0,0 +1,265 @@
+     * \{
+     */
+    //!\brief Inherit the parent class constructors.
+    using base_t::base_t;
+    //!\brief Inherit the parent class member function assign.
+    using base_t::assign;
+
+    //!\brief The default constructor initializes the underlying data member with '\0's.
+    constexpr small_string() noexcept : base_t{}
+    {
+        for (size_t i = 0; i < capacity_ + 1; ++i)
+            this->data_[i] = '\0';
+    }
+
+    /*!\brief Construction from literal.
+     * \param _lit The literal to construct the string for.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <size_t N>
+    explicit constexpr small_string(char const (&_lit)[N]) noexcept : small_string{}
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of small_string.");
+        assign(_lit);
+    }
+
+    /*!\brief Construction from char.
+     * \param c The character to construct the small_string for.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    explicit constexpr small_string(char const c) noexcept : small_string{}
+    {
+        assign(1, c);
+    }
+
+    /*!\brief Assign from literal.
+     * \param _lit The literal to assign the string from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <size_t N>
+    constexpr small_string & operator=(char const (&_lit)[N]) noexcept
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of small_string.");
+        assign(_lit);
+        return *this;
+    }
+
+    /*!\brief Assign from literal.
+     * \param _lit The literal to assign the string from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <size_t N>
+    constexpr void assign(char const (&_lit)[N]) noexcept
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of small_string.");
+        assert(_lit[N - 1] == '\0');
+        clear();
+        assign(&_lit[0], &_lit[N-1]);
+    }
+    //!\}
+
+    /*!\name Capacity
+     * \{
+     */
+    //!\brief Returns the maximal size which equals the capacity.
+    constexpr size_type max_size() const noexcept
+    {
+        return capacity_;
+    }
+
+    //!\brief Returns the maximal capacity.
+    constexpr size_type capacity() const noexcept
+    {
+        return capacity_;
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Returns the content represented as std::string.
+     *
+     * \returns `std::string` The stored string.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee. No data is modified.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the string.
+     */
+    std::string string() const
```

Comment:
Actually I would just call the member `.str()` and not `.string()`. The latter is really weird and the STL uses `.str()`...

Reply:
Also add an explicit conversion operator!

Reply:
it is called `str()` for streams but `string()` for filesystem paths.. but I'll change it to str

Reply:
> Also add an explicit conversion operator!

If we make it implicit, we can avoid the `.string()` in 
`get_display_name_v<me_type>.string()` :grin: